sdk/identity/identity/src/credentials/azureCliCredential.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { createSpan } from "../util/tracing";
+import { AuthenticationErrorName } from "../client/errors";
+import { CanonicalCode } from "@opentelemetry/types";
+import { logger } from "../util/logging";
+
+import * as child_process from "child_process";
+
+export interface CredentialClient {
+  getAzureCliAccessToken(resource: string): Promise<any>;
+}
+
+class AzureCliCredentialClient implements CredentialClient {
+  public getAzureCliAccessToken(resource: string) {
+    return new Promise((resolve, reject) => {
+      try {
+        child_process.exec(
+          `az account get-access-token --output json --resource ${resource}`,
+          (error, stdout, stderr) => {
+            resolve({ stdout: stdout, stderr: stderr });
+          }
+        );
+      } catch (err) {
+        reject(err);
+      }
+    });
+  }
+}
+
+export interface AzureCliCredentialOptions {
+  azureCliCredentialClient?: CredentialClient;
+}
+
+/**
+ * Provides the user access token and expire time
+ * with azure cli command "az account get-access-token" in powershell.
```

Comment:
Just some nits here on casing and removing an inaccuracy about it being invoked via PowerShell.

```suggestion
 * with Azure CLI command "az account get-access-token".
```

Reply:
Fixed